### DISCUSSION TITLE
Get master window using hyprctl?

### DISCUSSION THREAD

**Discussion by akiross** [hyprwm/Hyprland#6615] [Q&A] [answered]
I was interested in a feature like the one in issue #1265, so I was wondering whether it's possible to automate that using `hyprctl`, but to do that, I would need to get the id of the master window.

Is there a way to get the master window using `hyprctl`?

**Comment by Siri-chan** [comment, marked as answer]
You can do a surprising amount with `hyprctl -j` and `jq`, for example, you can write something like this:

```fish
hyprctl clients -j | jq "[.[] | select (.workspace.id == $(hyprctl activewindow -j | jq .workspace.id))] | min_by(.at[0]) | .address"
```
You might have to do some more filtering to handle floating/pseudotiled windows, or change the `min_by(.at[0])` to fit your master position (`max_by` will return the rightmost, and `[1]` will return a y-position), or some porting if you aren't using `fish`, but this should be a decent jumping off point.

**Reply by akiross** [reply]
Thank you very much!
